src/codeu/chat/server/View.java

```diff
@@ -181,6 +149,66 @@ public int conversationStatusUpdate(String title, Uuid owner) {
     return newMessages;
   }
 
+  private Collection<String> searchContributions(Time lastUpdate, Uuid searchUser) {
+
+    // Given a time value for the last time that a check for contributions was
+    // requested and the UUID of the user to search for recent contributions from,
+    // return a collection of the titles of conversations that the user has created
+    // or added messages to after the specified time.
+
+    Collection<String> contributions = new ArrayList<String>();
+    Collection<ConversationPayload> allConversations = all(model.conversationPayloadById());
+
+    // go through every conversation
+    for(ConversationPayload conversationPayload : allConversations) {
+      Message currentMessage = model.messageById().first(conversationPayload.firstMessage);
+      boolean foundMessage = false;
+      // go through every message
+      while(currentMessage != null && foundMessage == false) {
```

Comment:
Can this while loop be made into a separate method? The method will add all the messages to `contributions`.